app.rb

```diff
@@ -191,14 +195,14 @@ def admin_required!
 
   post "/change_password" do
     login_required!
-    if params[:oldpass] != current_user.password
+    if ( get_digested params[:oldpass] ) != current_user.password
       set_message "Your old password was entered incorrectly"
       redirect to "/change_password"
-    elsif params[:newpass] != params[:newpass2] || params[:newpass].length < 6
+    elsif ( params[:newpass] != params[:newpass2] ) || ( params[:newpass].length < 6 )
       set_message "Your passwords must match and be more than 6 characters"
       redirect to "/change_password"
     else
-      current_user.update password: params[:newpass]
+      current_user.update! password: (get_digested params[:newpass])
```

Comment:
An interesting take on this idea would be to override the `password=` method on the `User` object to set the _stored_ password to a digested version.